docs/devel/running-locally.md

```diff
@@ -1,449 +1,819 @@
-# Running Locally
-
-**Table of Contents**
-
-- [Requirements](#requirements)
-    - [OS](#os)
-    - [Docker](#docker)
-    - [etcd](#etcd)
-    - [Go](#go)
-    - [Node.js and NPM](#nodejs-and-npm)
-- [Clone the repository](#clone-the-repository)
-- [Build the binary](#build-the-binary)
-- [Make self-signed certificate](#make-self-signed-certificate)
-- [Running on the machine](#running-on-the-machine)
-    - [Create static token auth file](#create-static-token-auth-file)
-    - [tke-auth](#tke-auth)
-    - [tke-platform-api](#tke-platform-api)
-    - [tke-platform-controller](#tke-platform-controller)
-    - [tke-business-api](#tke-business-api)
-    - [tke-business-controller](#tke-business-controller)
-    - [tke-gateway](#tke-gateway)
-- [Open UI console](#open-ui-console)
-
-## Requirements
-
-### OS
-
-TKE support running on `linux`, `Windows` or `macOS` system.
+# How to run TKE locally
+This guide will walk you through deploying the full TKE stack on you local machine and allow you to play with the core components. It is highly recommended if you want to develop TKE and contribute regularly.
+
+## Table of Contents
+
+- [Prerequisites](#prerequisites)
+  - [OS Requirements](#os-requirements)
+  - [Docker](#docker)
+  - [etcd](#etcd)
+  - [Go](#go)
+  - [Node.js and NPM](#node-js-and-npm)
+- [Building TKE Components](#building-tke-components)
+- [Create Self-signed Certificates](#create-self-signed-certificates)
+- [Create Static Token](#create-static-token)
+- [Bootstrap TKE Core Components](#bootstrap-tke-core-components)
+  - [tke-auth](#tke-auth)
+  - [tke-platform-api](#tke-platform-api)
+  - [tke-platform-controller](#tke-platform-controller)
+  - [tke-registry-api(Optional)](#tke-registry-api-optional-)
```

Comment:
tke-registry-api seems not an optional component at installation process. tke-installer will push images to registry through registry-api after tke components are deployed.

Reply:
Thanks for the response. 

They are different processes to launch TKE. This "running locally" guide covers the process of running TKE core components one by one on local machine for development purpose while the `tke-installer` gives you an easy way to install TKE stack and includes some of our best practice. We made the "running locally" process more flexible and you can choose whether you want to enable `tke-registry-api`.